src/backend/distributed/master/master_node_protocol.c

```diff
@@ -482,10 +485,128 @@ GetTableDDLEvents(Oid relationId, bool includeSequenceDefaults)
 	indexAndConstraintCommandList = GetTableIndexAndConstraintCommands(relationId);
 	tableDDLEventList = list_concat(tableDDLEventList, indexAndConstraintCommandList);
 
+	replicaIdentityEvents = GetTableReplicaIdentityCommand(relationId);
+	tableDDLEventList = list_concat(tableDDLEventList, replicaIdentityEvents);
+
 	return tableDDLEventList;
 }
 
 
+static List *
+GetTableReplicaIdentityCommand(Oid relationId)
+{
+	List *replicaIdentityCreateCommandList = NIL;
+	Relation pgClass = NULL;
+	SysScanDesc scanDescriptor = NULL;
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 1;
+	HeapTuple heapTuple = NULL;
+	OverrideSearchPath *overridePath = NULL;
+
+	char relationKind = get_rel_relkind(relationId);
+	if (relationKind != RELKIND_RELATION)
+	{
+		return NIL;
+	}
+
+	/*
+	 * Set search_path to NIL so that all objects outside of pg_catalog will be
+	 * schema-prefixed. pg_catalog will be added automatically when we call
+	 * PushOverrideSearchPath(), since we set addCatalog to true;
+	 */
+	overridePath = GetOverrideSearchPath(CurrentMemoryContext);
+	overridePath->schemas = NIL;
+	overridePath->addCatalog = true;
+	PushOverrideSearchPath(overridePath);
+
+	/* open system catalog and find the table with the given relationId */
+	pgClass = heap_open(RelationRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_index_indrelid,
+				BTEqualStrategyNumber, F_OIDEQ, relationId);
+
+	ScanKeyInit(&scanKey[0],
+				ObjectIdAttributeNumber,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(relationId));
+
+	scanDescriptor = systable_beginscan(pgClass, ClassOidIndexId, true,
+										NULL, scanKeyCount, scanKey);
+
+	heapTuple = systable_getnext(scanDescriptor);
+
+	if (HeapTupleIsValid(heapTuple))
+	{
+		Form_pg_class pgClassForm = (Form_pg_class) GETSTRUCT(heapTuple);
+		char replicaIdentity = pgClassForm->relreplident;
+
+		AppendReplicaIdentityCommands(&replicaIdentityCreateCommandList, replicaIdentity,
+									  relationId);
+	}
+
+	/* clean up scan and close system catalog */
+	systable_endscan(scanDescriptor);
+	heap_close(pgClass, AccessShareLock);
+
+	/* revert back to original search_path */
+	PopOverrideSearchPath();
+
+	return replicaIdentityCreateCommandList;
+}
+
+
+/*
+ * AppendReplicaIdentityCommands function appends the required DDL commands
+ * to the replicaIdentityCreateCommandList.
+ */
+static void
+AppendReplicaIdentityCommands(List **replicaIdentityCreateCommandList,
+							  char replicaIdentity, Oid relationId)
+{
+	StringInfo buf = NULL;
+	char *relationName = NULL;
+
+	buf = makeStringInfo();
+	relationName = generate_qualified_relation_name(relationId);
```

Comment:
Might be worth splitting up this function into one that takes an OID and that takes a relation name. That way we could use this with shard names in the future.